### ISSUE TITLE
Menu has incorrect ARIA roles

### ISSUE THREAD

**Issue by amlehman** [mantinedev/mantine#5372]
### What package has an issue?

@mantine/core

### Describe the bug

The Menu and MenuDropdown components fail a WCAG/508 accessibility scan with "Certain ARIA roles must contain particular children". There is an extra div wrapping the elements with `role=menuitem` causing them not to be direct descents of the element with `role=menu` as they should be. 

<img width="759" alt="Screenshot 2023-12-05 at 9 49 21 AM" src="https://github.com/mantinedev/mantine/assets/4624053/458758fe-771e-4a05-a526-d54d26937974">

### What version of @mantine/* packages do you have in package.json? (Note that all @mantine/* packages must have the same version in order to work correctly)

6.0.21

### Do you know how to fix the issue

None

### Do you want to send a pull request with a fix?

None

### Possible fix

Remove extra div or add `role=group` to it, example: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/group_role#examples

**Comment by amlehman** [comment]
Hi, I'm still seeing the issue present in 7.3.1. I'm also not seeing anything in that pull request that would have addressed the `menuitem` elements not having a parent element with a valid role.

**Comment by rtivital** [comment]
Fixed in [7.3.2](https://github.com/mantinedev/mantine/releases/tag/7.3.2)

**Comment by vijayprasanna13** [comment]
I'm seeing this in `7.4.0`. Is there a workaround available?

![Screenshot 2024-02-20 at 4 27 13 PM](https://github.com/mantinedev/mantine/assets/11921040/79e209a7-017e-4d78-abf6-96c860c879df)
